GrafanaStream/README.md

```diff
@@ -0,0 +1,211 @@
+- The client_writer writes the hosts stories (CPU, Memory, Network) into ChronoLog
+- The client_reader_stream_influx reads these stories from ChronoLog and streams to InfluxDB
+- The Grafana dashboard visualizes the metrics in Grafana for the particular Chronicle and Story
+
+### 1. Scripts
+#### ClientScripts/client_writer.cpp
+- **Purpose:** This is a simple client producer that writes CPU/Memory/Network metrics to chronolog
+- **How it works:** It reads real metrics from /proc, acquires/creates three stories (cpu_usage, memory_usage, network_usage), and logs values for every interval_sec
+
+#### ClientScripts/client_reader_stream_influx.cpp
+- **Purpose:** This script streams the events from ChronoLog to InfluxDB for Grafana
+- **How it works:** It connects to chronolog, ensures the chronicle/stories exist, then on a loop calls the ReplayStory(t_start, t_end) api. The events are converted to line protocol through the transform api created and sent in batches via InfluxDBSink. It also maintains a pe -story cursor to avoid re reading the events
+
+#### StreamingScripts/InfluxDBSink.h and InfluxDBSink.cpp
+- **Purpose:** This is the HTTP sink that posts the TelemetryBatch to InfluxDB v2
+- **How it works:** It Uses libcurl to POST line protocol to the configured write URL (org, bucket, precision=ns). It supports token auth, timeouts and retries
+
+#### StreamingScripts/StreamSink.h
+- **Purpose:** This is the interface used for the streaming backends, it is generic so it can be re-used for api v2
+
+#### StreamingScripts/Transform.h and Transform.cpp
+- **Purpose:** This converts the chronolog events into TelemetryBatch customised for the influx sink
+- **How it works:** It parses the payloads per story, adds tags (chronicle, story) and attaches the event timestamp (ns). It produces properly escaped line protocol ready for Influx
+
+#### GrafanaInfluxSetup/
+- **Purpose:** This contains all the references for setting up Grafana and Influx
+- The docker compose is for launching the containers
+- The .env defines the env variables needed by the setup yml scripts
+- The dashboards/..json is the json used for templating and auto provisioning the Grafana dashboard on start up so no manual intervention is needed on the UI
+- The provisioning/datasources/datasource.yml has all the details for provisioning the influxdb data source
+- The provisioning/dashboards/chronolog/yml has all the details for provisioning grafana 
+
+### 2. Environment setup
+Launch the below commands on your terminal to get the session ready
+``` bash
+cd Chronolog
+source ~/<directory where spack is cloned>/spack/share/spack/setup-env.sh
+spack env activate -p .
+
+# Get your VM IP
+hostname -I | awk '{print $1}'
+# Set that as an env variable
+export VM_IP=192.168.64.6
+export INSTALL_PREFIX=~/chronolog/Debug/
+export SRC_ROOT=~/Desktop/ChronoLog
+export LD_LIBRARY_PATH=$INSTALL_PREFIX/lib:$SRC_ROOT/.spack-env/view/lib:$SRC_ROOT/.spack-env/view/lib64:$LD_LIBRARY_PATH
```

Comment:
pls use `$INSTALL_PREFIX/lib` instead of spack's lib directory. an even better option is to set `LD_LIBRARY_PATH` everything temporarily before running any executables.